--- contrib/libs/onnx_runtime/onnxruntime/core/platform/posix/stacktrace.cc	(index)
+++ contrib/libs/onnx_runtime/onnxruntime/core/platform/posix/stacktrace.cc	(working tree)
@@ -13,7 +13,7 @@ namespace onnxruntime {
 std::vector<std::string> GetStackTrace() {
   std::vector<std::string> stack;
 
-#if !defined(NDEBUG) && !defined(__ANDROID__) && !defined(__wasm__) && !defined(_OPSCHEMA_LIB_)
+#if 0
   constexpr int kCallstackLimit = 64;  // Maximum depth of callstack
 
   void* array[kCallstackLimit];
